src/main/java/yahav/photomosaic/ImagePixels.java

```diff
@@ -19,12 +20,12 @@ public ImagePixels() throws IOException {
     }
 
 
-    public int[][] getImagePixels(Image image, int width, int height) {
+    public int[] getImagePixels(Image image, int width, int height) {
         PixelReader pixelReader = image.getPixelReader();
-        int[][] pixels = new int[width][height];
+        int[] pixels = new int[width*height];
         for(int row = 0; row < height; row++) {
             for (int col = 0; col < width; col++) {
-                pixels[row][col] = image.getPixelReader().getArgb(row,col);
+                pixelReader.getPixels(row,col,width,height, PixelFormat.getIntArgbInstance(),pixels,0,width);
```

Comment:
I'm not really understanding the getPixel format from javafx, this gives me an error.